pkg/awsclient/route53_helper.go

```diff
@@ -1,38 +1,44 @@
 package awsclient
 
 import (
-	"strings"
+	"path"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/route53"
 )
 
-// UpsertARecord adds an A record alias named DNSName in the target zone aliasDNSZoneID, inside the clusterDomain's zone.
-func (c *AwsClient) UpsertARecord(clusterDomain, DNSName, aliasDNSZoneID, resourceRecordSetName, comment string, targetHealth bool) error {
-	// look up clusterDomain to get hostedzoneID
-	lookup := &route53.ListHostedZonesByNameInput{
+// GetPublicHostedZoneID looks up the ID of the public hosted zone for clusterDomain.
+func (c *AwsClient) GetPublicHostedZoneID(clusterDomain string) (string, error) {
+	input := &route53.ListHostedZonesByNameInput{
 		DNSName: aws.String(clusterDomain),
 	}
-
-	listHostedZones, err := c.ListHostedZonesByName(lookup)
+	output, err := c.ListHostedZonesByName(input)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	// get public hosted zone ID needed to changeResourceRecordSets
 	var publicHostedZoneID string
-	for _, zone := range listHostedZones.HostedZones {
+	for _, zone := range output.HostedZones {
 		if *zone.Name == clusterDomain {
-			// In order to get the publicHostedZoneID we need to get
-			// the HostedZone.Id object which is in the form of "/hostedzone/Z1P3C0HZA40C0N"
-			// Since we only care about the ID number, we take index of the last "/" char and parse right
-			zoneID := aws.StringValue(zone.Id)
-			slashIndex := strings.LastIndex(zoneID, "/")
-			publicHostedZoneID = zoneID[slashIndex+1:]
+			// The zone ID is the last element of the string
+			// HostedZone.Id, which takes the form of a path:
+			// "/hostedzone/<ZONEID>"
+			publicHostedZoneID = path.Base(aws.StringValue(zone.Id))
```

Comment:
I'm not sure about this change with using path package in this way. Path seems to want to be used with files and URLs. Is it coincidence that these hosted zones take that form? There's no doubt that it works but it's a counterintuitive use, eh?

Reply:
As I understand it, [path.filepath](https://golang.org/pkg/path/filepath/) is for files but [path](https://golang.org/pkg/path/filepath/) is for any kind of slash-separated identifier.

Hosted zone IDs _happen_ to be path-like, and as long as we're using that knowledge to pick it apart, [path](https://golang.org/pkg/path/filepath/) seems to me like an appropriate use.

Reply:
The note from the `path` [package docs](https://pkg.go.dev/path?tab=doc):

> The path package should only be used for paths separated by forward slashes, such as the paths in URLs. This package does not deal with Windows paths with drive letters or backslashes; to manipulate operating system paths, use the [path/filepath](https://pkg.go.dev/path/filepath@go1.14.6?tab=doc) package.

I don't think this is a particularly important hill for me to die on, but I want to be kind to our future selves with idiomatic use of language packages. Maybe someone with more experienced Go chops than me could weigh in? @cblecker maybe?

Reply:
Amazon's Route 53 API for making changes to hosted zones is designed as a restful API. The ID [returned here](https://docs.aws.amazon.com/Route53/latest/APIReference/API_HostedZone.html) is a URI path, and IMO is an appropriate use of the path package based on:
> The path package should only be used for paths separated by forward slashes, such as the paths in URLs.

Reply:
Works for me 👍

Reply:
+1 this looks a lot cleaner than before. Since we are using it in the other controllers though it would be good to test and make sure everything works as expected

Reply:
I've tested it in the context of the SSHD controller and it works fine afaict.